pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/LLCSegmentCompletionHandlers.java

```diff
@@ -260,13 +291,13 @@ public String segmentUpload(@QueryParam(SegmentCompletionProtocol.PARAM_INSTANCE
     requestParams.withInstanceId(instanceId).withSegmentName(segmentName).withOffset(offset);
     LOGGER.info("Processing segmentUpload:{}", requestParams.toString());
 
-    final String segmentLocation = uploadSegment(multiPart, instanceId, segmentName, true);
-    if (segmentLocation == null) {
+    final ImmutablePair<URI, SegmentMetadataImpl> uploadResults = uploadSegment(multiPart, instanceId, segmentName, true);
```

Comment:
we should not be extracting metadata in this handler, since we are doing that when the commit end happens.

Reply:
No. I am not doing metadata extracting in this handler. There is a control flag isSplitCommit in calling the uploadSegment(). If it is true, there will be no metadata extraction. Note that we pass true at this line.